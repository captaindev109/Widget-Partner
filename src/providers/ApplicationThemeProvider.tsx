import { ThemeProvider, DefaultTheme } from 'styled-components';
import GlobalStyles from '@/styles/Global';
import React from 'react';
import { ChildrenReturnType } from '@/types/types';

interface ApplicationThemeProviderProps {
  children?: ChildrenReturnType;
  theme: DefaultTheme;
}

const ApplicationThemeProvider = ({ children, theme }: ApplicationThemeProviderProps) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>
  );
};

export default ApplicationThemeProvider;
