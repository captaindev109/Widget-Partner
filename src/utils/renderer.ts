import React from 'react';
import { ConfigProps, Index, EachComponentObjectChildren, BasicAny } from '@/types/types';
import { UIComponents } from './components';

const recursivelyCreateReactComponents = (
  componentToCreate: BasicAny,
  configuration: ConfigProps,
  children: BasicAny
) => {
  if (typeof componentToCreate !== 'undefined') {
    return React.createElement(
      componentToCreate,
      configuration,
      children &&
        (typeof children === 'string'
          ? children
          : children.map((component: EachComponentObjectChildren, index: Index) => viewRenderer(component, index)))
    );
  }
  null;
};

const viewRenderer: BasicAny = (config: ConfigProps, index: Index) => {
  const children = config.children;
  const newConfig = { ...config, key: index };
  const LookForComponentInUILibrary = UIComponents[config.component];
  return recursivelyCreateReactComponents(LookForComponentInUILibrary, newConfig, children);
};

export default viewRenderer;
