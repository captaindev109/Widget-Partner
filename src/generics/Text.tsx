import { device } from '@/utils/constants';
import styled from 'styled-components';
import { BasicAny, TextProps } from '@/types/types';

const StyledText = styled.p<TextProps>`
  text-align: ${({ textAlign }) => textAlign};
  color: ${(props) => props.color && props.theme.color[props.color]};
  font-family: ${(props) => props.theme.fonts.FAVORIT};
  font-weight: ${(props) => props.weight};
  @media ${device.mobileS} {
    ${(props) => {
      switch (props.variant) {
        case 'HeadlineXL':
          return `
            font-size: ${props.theme.size.fontSizes.f10};
            line-height: ${props.theme.size.lineHeights.lh12};
            letter-spacing: ${props.theme.size.letterSpacings.lsp2};
          `;
        case 'HeadlineL':
          return `
            font-size: ${props.theme.size.fontSizes.f8};
            line-height: ${props.theme.size.lineHeights.lh10};
            letter-spacing: ${props.theme.size.letterSpacings.lsp2};
          `;
        case 'HeadlineM':
          return `
            font-size: ${props.theme.size.fontSizes.f7};
            line-height: ${props.theme.size.lineHeights.lh9};
            letter-spacing: ${props.theme.size.letterSpacings.lsp2};
          `;
        case 'HeadlineS':
          return `
            font-size: ${props.theme.size.fontSizes.f6};
            line-height: ${props.theme.size.lineHeights.lh8};
            letter-spacing: ${props.theme.size.letterSpacings.lsp1};
          `;
        case 'HeadlineXS':
          return `
            font-size: ${props.theme.size.fontSizes.f5};
            line-height: ${props.theme.size.lineHeights.lh6};
            letter-spacing: ${props.theme.size.letterSpacings.ls1};
          `;
        case 'OVERLINE':
          return `
            font-size: ${props.theme.size.fontSizes.f2};
            line-height: ${props.theme.size.lineHeights.lh3};
            letter-spacing: ${props.theme.size.letterSpacings.ls3};
          `;
        case 'Body1':
          return `
            font-size: ${props.theme.size.fontSizes.f4};
            line-height: ${props.theme.size.lineHeights.lh5};
            letter-spacing: ${props.theme.size.letterSpacings.ls1};
          `;
        case 'Body2':
          return `
            font-size: ${props.theme.size.fontSizes.f3};
            line-height: ${props.theme.size.lineHeights.lh4};
            letter-spacing: ${props.theme.size.letterSpacings.ls1};
          `;
        case 'Body3':
          return `
            font-size: ${props.theme.size.fontSizes.f2};
            line-height: ${props.theme.size.lineHeights.lh2};
            letter-spacing: ${props.theme.size.letterSpacings.ls1};
          `;
        case 'Pullquote':
          return `
            font-family: ${props.theme.fonts.EB_GARAMOND};
            font-size: ${props.theme.size.fontSizes.f7};
            line-height: ${props.theme.size.lineHeights.lh9};
            letter-spacing: ${props.theme.size.letterSpacings.ls2};
          `;
        case 'Caption':
        default:
          return `
            font-size: ${props.theme.size.fontSizes.f1};
            line-height: ${props.theme.size.lineHeights.lh1};
            letter-spacing: ${props.theme.size.letterSpacings.ls1};
          `;
      }
    }}
  }
  @media ${device.laptop} {
    ${(props) => {
      switch (props.variant) {
        case 'HeadlineXL':
          return `
            font-size: ${props.theme.size.fontSizes.f11};
            line-height: ${props.theme.size.lineHeights.lh13};
            letter-spacing: ${props.theme.size.letterSpacings.lsp2};
          `;
        case 'HeadlineL':
          return `
            font-size: ${props.theme.size.fontSizes.f10};
            line-height: ${props.theme.size.lineHeights.lh12};
            letter-spacing: ${props.theme.size.letterSpacings.lsp2};
          `;
        case 'HeadlineM':
          return `
            font-size: ${props.theme.size.fontSizes.f9};
            line-height: ${props.theme.size.lineHeights.lh11};
            letter-spacing: ${props.theme.size.letterSpacings.lsp2};
          `;
        case 'HeadlineS':
          return `
            font-size: ${props.theme.size.fontSizes.f7};
            line-height: ${props.theme.size.lineHeights.lh9};
            letter-spacing: ${props.theme.size.letterSpacings.lsp2};
          `;
        case 'HeadlineXS':
          return `
            font-size: ${props.theme.size.fontSizes.f6};
            line-height: ${props.theme.size.lineHeights.lh8};
            letter-spacing: ${props.theme.size.letterSpacings.lsp2};
          `;
        case 'OVERLINE':
          return `
            font-size: ${props.theme.size.fontSizes.f3};
            line-height: ${props.theme.size.lineHeights.lh4};
            letter-spacing: ${props.theme.size.letterSpacings.ls1};
          `;
        case 'Body1':
          return `
            font-size: ${props.theme.size.fontSizes.f4};
            line-height: ${props.theme.size.lineHeights.lh5};
            letter-spacing: ${props.theme.size.letterSpacings.ls1};
          `;
        case 'Body2':
          return `
            font-size: ${props.theme.size.fontSizes.f3};
            line-height: ${props.theme.size.lineHeights.lh4};
            letter-spacing: ${props.theme.size.letterSpacings.ls1};
          `;
        case 'Body3':
          return `
            font-size: ${props.theme.size.fontSizes.f2};
            line-height: ${props.theme.size.lineHeights.lh2};
            letter-spacing: ${props.theme.size.letterSpacings.ls1};
          `;
        case 'Pullquote':
          return `
            font-family: ${props.theme.fonts.EB_GARAMOND};
            font-size: ${props.theme.size.fontSizes.f9};
            line-height: ${props.theme.size.lineHeights.lh11};
            letter-spacing: ${props.theme.size.letterSpacings.ls2};
          `;
        case 'Caption':
        default:
          return `
            font-size: ${props.theme.size.fontSizes.f1};
            line-height: ${props.theme.size.lineHeights.lh1};
            letter-spacing: ${props.theme.size.letterSpacings.ls1};
          `;
      }
    }}
  }
`;

const Text = ({ children, ...newProps }: BasicAny) => {
  return <StyledText {...newProps}>{children}</StyledText>;
};

Text.defaultProps = {
  textAling: 'left',
  color: 'gray900',
  weight: 'normal',
};

export default Text;
