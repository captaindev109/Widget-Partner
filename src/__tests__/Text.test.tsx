import { render, RenderResult } from '@testing-library/react';
import queries from '@testing-library/dom/types/queries';
import renderer from 'react-test-renderer';
import Text from '@/generics/Text';
import ApplicationThemeProvider from '@/providers/ApplicationThemeProvider';
import defaultTheme from '@/theme/theme';

const children = 'We offer insurance';

const MockComponent: React.FC = () => {
  return (
    <ApplicationThemeProvider theme={defaultTheme}>
      <Text variant="HeadlineXL">{children}</Text>
    </ApplicationThemeProvider>
  );
};

describe('Test cases of the Text component', () => {
  let renderedComponent: RenderResult<typeof queries, HTMLElement>;

  beforeEach(() => {
    renderedComponent = render(<MockComponent />);
  });

  test('to see if the component render without crashing', () => {
    render(<MockComponent />);
  });

  test('to see if the description text is correct', () => {
    const { getByText } = renderedComponent;
    expect(getByText(children)).toBeInTheDocument();
  });

  it('Snapshot test should pass', () => {
    const snap = renderer.create(<MockComponent />).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
