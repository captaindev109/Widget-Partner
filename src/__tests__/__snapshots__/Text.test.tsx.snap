// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases of the Text component Snapshot test should pass 1`] = `
<p
  className="Text__StyledText-sc-1jfsuh1-0 gBzvyB"
  color="gray900"
>
  We offer insurance
</p>
`;
